src/server/game/Movement/MovementGenerators/TargetedMovementGenerator.cpp

```diff
@@ -33,7 +33,7 @@ void TargetedMovementGeneratorMedium<T, D>::_setTargetLocation(T* owner, bool up
     if (!i_target.isValid() || !i_target->IsInWorld())
         return;
 
-    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE))
+    if (owner->HasUnitState(UNIT_STATE_NOT_MOVE | UNIT_STATE_CASTING))
```

Comment:
@Treeston This PR add only a missing check. UNIT_STATE_CASTING is checked on every update in TargetedMovementGenerator. So to support something like this many more parts needs to be changed. TC has generally not a support for casting during moving. TL:DR This change dosent break casting while moving because it wouldnt work in the first place :)